database/mssql/scripts/dbscripts/PSP_PIMS_S24_00/Alter Down/01_PSP_PIMS_S24_00_Alter_Down.sql

```diff
@@ -0,0 +1,3335 @@
+	DROP CONSTRAINT [RESRCH_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop primary key constraint dbo.RSRCHS_PK
+PRINT N'Drop primary key constraint dbo.RSRCHS_PK'
+GO
+ALTER TABLE [dbo].[PIMS_RESEARCH_FILE_STATUS_TYPE]
+	DROP CONSTRAINT [RSRCHS_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop primary key constraint dbo.BCAJUR_PK
+PRINT N'Drop primary key constraint dbo.BCAJUR_PK'
+GO
+ALTER TABLE [dbo].[BCA_JURISDICTION]
+	DROP CONSTRAINT [BCAJUR_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop primary key constraint dbo.BCASAR_PK
+PRINT N'Drop primary key constraint dbo.BCASAR_PK'
+GO
+ALTER TABLE [dbo].[BCA_ASSESSMENT_AREA]
+	DROP CONSTRAINT [BCASAR_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop primary key constraint dbo.VOLUTY_PK
+PRINT N'Drop primary key constraint dbo.VOLUTY_PK'
+GO
+ALTER TABLE [dbo].[PIMS_VOLUME_UNIT_TYPE]
+	DROP CONSTRAINT [VOLUTY_PK]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Drop sequence dbo.PIMS_RFILE_NUMBER_SEQ
+PRINT N'Drop sequence dbo.PIMS_RFILE_NUMBER_SEQ'
+GO
+DROP SEQUENCE [dbo].[PIMS_RFILE_NUMBER_SEQ]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Alter table dbo.PIMS_PROPERTY
+PRINT N'Alter table dbo.PIMS_PROPERTY'
+GO
+ALTER TABLE [dbo].[PIMS_PROPERTY] ALTER COLUMN [PROPERTY_AREA_UNIT_TYPE_CODE] nvarchar(20) NOT NULL
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+ALTER TABLE [dbo].[PIMS_PROPERTY]
+	ADD [PROPERTY_TENURE_TYPE_CODE] nvarchar(20) NOT NULL, 
+	[VOLUMETRIC_UNIT_TYPE_CODE] nvarchar(20) NULL, 
+	[PROPERTY_ANOMALY_TYPE_CODE] nvarchar(20) NULL, 
+	[PROPERTY_ROAD_TYPE_CODE] nvarchar(20) NULL, 
+	[PROPERTY_ADJACENT_LAND_TYPE_CODE] nvarchar(20) NULL
+GO
+EXEC sp_addextendedproperty 
+	@name = N'MS_Description', @value = N'Volumetric measurement unit of the parcel.' , 
+	@level0type = N'Schema', @level0name = N'dbo', 
+	@level1type = N'Table', @level1name = N'PIMS_PROPERTY', 
+	@level2type = N'Column', @level2name = N'VOLUMETRIC_UNIT_TYPE_CODE'
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+ALTER TABLE [dbo].[PIMS_PROPERTY]
+	DROP COLUMN [VOLUME_UNIT_TYPE_CODE]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Alter table dbo.PIMS_PROPERTY_HIST
+PRINT N'Alter table dbo.PIMS_PROPERTY_HIST'
+GO
+ALTER TABLE [dbo].[PIMS_PROPERTY_HIST] ALTER COLUMN [PROPERTY_AREA_UNIT_TYPE_CODE] nvarchar(20) NOT NULL
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+ALTER TABLE [dbo].[PIMS_PROPERTY_HIST]
+	ADD [PROPERTY_TENURE_TYPE_CODE] nvarchar(20) NOT NULL, 
+	[VOLUMETRIC_UNIT_TYPE_CODE] nvarchar(20) NULL, 
+	[PROPERTY_ANOMALY_TYPE_CODE] nvarchar(20) NULL, 
+	[PROPERTY_ROAD_TYPE_CODE] nvarchar(20) NULL, 
+	[PROPERTY_ADJACENT_LAND_TYPE_CODE] nvarchar(20) NULL
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+ALTER TABLE [dbo].[PIMS_PROPERTY_HIST]
+	DROP COLUMN [VOLUME_UNIT_TYPE_CODE]
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Alter table dbo.BCA_ASSESSMENT_AREA
+PRINT N'Alter table dbo.BCA_ASSESSMENT_AREA'
+GO
+ALTER TABLE [dbo].[BCA_ASSESSMENT_AREA]
+	ADD CONSTRAINT [PK__BCA_ASSE__FC7CF8094B7DBA6B]
```

Comment:
uh oh, why are there unnamed constraints here? this means that in our other environments we may have constraints with different generated names correct? We will likely need a ticket to go clean those up. (thankfully I think BC assessment was just from last sprint so this should only be an issue in test.

Reply:
Fixed now